Dockerfile

```diff
@@ -5,17 +5,38 @@ FROM ubuntu:23.10
 
 RUN apt-get update --yes >/dev/null && \
     apt-get install --yes -qq \
+        curl \
         python3 \
         python3-ahocorasick \
         python3-bpfcc \
+        python3-docker \
         python3-prometheus-client \
         python3-structlog \
         python3-psutil \
+        tar \
         tini \
         bpftrace && \
     apt-get clean && \
     rm -rf /var/lib/apt/lists/*
 
+ARG CRICTL_VERSION=1.29.0
+RUN MACHINE=`uname -m`; \
+    if [ "$MACHINE" = "x86_64" ]; then \
+        ARCH=amd64; \
+    elif [ "$MACHINE" = "aarch64" ]; then \
+        ARCH=arm64; \
+    else \
+        echo "Unsupported architecture: $MACHINE"; \
+        exit 1; \
+    fi; \
+    curl -sSfL https://github.com/kubernetes-sigs/cri-tools/releases/download/v${CRICTL_VERSION}/crictl-v${CRICTL_VERSION}-linux-${ARCH}.tar.gz | \
+        tar -C /usr/local/bin -xzf -
+
+# These must match the values in daemonset.yaml
+ENV CONTAINER_RUNTIME_ENDPOINT=unix:///run/containerd/containerd.sock
```

Comment:
Let's leave these unset and error at runtime if they are not set? Easier to debug I think.